cmd/openfga/openfga_tls_test.go

```diff
@@ -0,0 +1,208 @@
+VtDSuN4wv0/DYztNdMwoSdWxELrcLxv4VQR7VV0KcUAQQBm4A+DvJ4t4HI1m6mED
+TOHJhohDqqoBvtpwC0gjOCzn2gh6lRSXsllTA2ECgYEAmw/pz1KKFt4Z/QvmwfMa
+Ys3vUy0wmfksgf9SqSK1oGn32ofXUFbR2peW3pqLp/ZTUIzHfR+WBAeKQ312Tqm0
+4wFwtrpISgR1OmdNeluG6PhMWbBUHcp3XknEFh0cc5RqBO5aeoTcXM4WccV+wFck
+sApBeBhCzjAZzr/fCquQS6MCgYAxA5az9LojpBrfrgh2hLRK+5QGzkCrXZi5EOSZ
+jktiYc/Te1ogL4c+IVsGi+eoWFRc0w8jsJY0i0Rte0W2elyrRNZphEWu8OdSbcBl
+BRs5IefJwzNFyvfKp3ztCBZdCC6djyU2pizLkje4q8qmSrjoV9AkSIlkhq8OxHIl
+D5s/YQKBgQCzflz985M6k1KT5EG/dopUUqgWJh1h07A275q9DswBl+XmDBAVdmiB
+JgSb5y84gQNFxuvXT1ZXV/lQytB6ZpCxYS1gZybQAUGlvPKQGW5NoXqDucimPpvm
+p2CIcm9oUH1iG9P/ELWz/it0RUbfy7GuBsUJn9MOdiO58uTiDBE7Kg==
+-----END RSA PRIVATE KEY-----`
+)
+
+func createKeys(t *testing.T) {
+	certFile, err := ioutil.TempFile("", "server_cert")
+	require.NoError(t, err, "error creating pem file")
+	defer certFile.Close()
+
+	_, err = certFile.Write([]byte(serverCert))
+	require.NoError(t, err, "failed to write pem")
+
+	keyFile, err := ioutil.TempFile("", "server_key")
+	require.NoError(t, err, "error creating pem file")
+	defer keyFile.Close()
+
+	_, err = keyFile.Write([]byte(serverKey))
+	require.NoError(t, err, "failed to write pem")
+
+	require.NoError(t, os.Setenv(tlsEnableEnvVar, "true"), "failed to set env var")
+	require.NoError(t, os.Setenv(tlsCertPathEnvVar, certFile.Name()), "failed to set env var")
+	require.NoError(t, os.Setenv(tlsKeyPathEnvVar, keyFile.Name()), "failed to set env var")
+}
+
+func TestServingTLS(t *testing.T) {
+	t.Run("failing to set cert path will not allow server to start", func(t *testing.T) {
+		require.NoError(t, os.Setenv(tlsEnableEnvVar, "true"), "failed to set env var")
+		require.NoError(t, os.Setenv(tlsKeyPathEnvVar, "some/path"), "failed to set env var")
+		defer os.Clearenv()
+
+		_, err := buildService(logger.NewNoopLogger())
+		require.ErrorIs(t, err, errInvalidTLSConfig)
+	})
+
+	t.Run("failing to set key path will not allow server to start", func(t *testing.T) {
+		require.NoError(t, os.Setenv(tlsEnableEnvVar, "true"), "failed to set env var")
+		require.NoError(t, os.Setenv(tlsCertPathEnvVar, "some/path"), "failed to set env var")
+		defer os.Clearenv()
+
+		_, err := buildService(logger.NewNoopLogger())
+		require.ErrorIs(t, err, errInvalidTLSConfig)
+	})
+
+	t.Run("Enable TLS is false, even with keys set, will serve plaintext", func(t *testing.T) {
+		createKeys(t)
+		require.NoError(t, os.Setenv(tlsEnableEnvVar, "false"), "failed to set env var") // override
+		defer os.Clearenv()
+
+		service, err := buildService(logger.NewNoopLogger())
+		require.NoError(t, err)
+		defer service.openFgaServer.Close()
+
+		ctx := context.Background()
+		g, ctx := errgroup.WithContext(ctx)
+		g.Go(func() error {
+			return service.openFgaServer.Run(ctx)
+		})
+
+		ensureServiceUp(t)
+	})
+
+	t.Run("Enable TLS is true will serve TLS", func(t *testing.T) {
+		createKeys(t)
+		require.NoError(t, os.Setenv("OPENFGA_HTTP_PORT", "9090"), "failed to set env var")
```

Comment:
Why do we need to set this? Can't we omit this and leave it at it's default value (e.g. 8080)? It would clean the test up a bit 🤷